apps/homeless-people-help-map/src/components/App.tsx

```diff
@@ -123,17 +125,34 @@ export const App = () => {
     return e;
   }, [districtFilter.expression, layerFilterFixedExpression]);
 
+  const handleSideBar = useCallback((value: boolean, changePrevious: boolean) => {
+    if (changePrevious) frameStateSideBar.current = value;
+    setSidebarVisible(value);
+  }, []);
+
   // close sidebar on mobile and open on desktop
   useEffect(() => {
     // from mobile to desktop
-    if (previousMobile !== false && isMobile === false) {
-      setSidebarVisible(true);
+    if (previousMobile == true && !isMobile) {
+      handleSideBar(true, true);
     }
     // from desktop to mobile
-    if (previousMobile !== true && isMobile === true) {
-      setSidebarVisible(false);
+    if (previousMobile == false && isMobile) {
+      handleSideBar(false, true);
+    }
+
+    window === window.parent || isMobile ? setFrameState(false) : setFrameState(true);
```

Comment:
add comment detailing hat this does and why (I guess the first part is first is "is not inside iframe")